sympy/matrices/expressions/dotproduct.py

```diff
@@ -0,0 +1,56 @@
+from __future__ import print_function, division
+
+from sympy.core import Basic, sympify
+from sympy.matrices.expressions.matexpr import (MatrixExpr)
+
+
+class DotProduct(MatrixExpr):
+    """
+        Dot Product of vector matrices
+    """
+
+    def __new__(cls, *args):
```

Comment:
Instead of checking the number of args in the validate function, just make this `__new__(cls, arg1, arg2)`.